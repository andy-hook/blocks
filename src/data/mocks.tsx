import { SocialMeta } from "@custom-types/model"
import { FluidObject } from "gatsby-image"
import { TransitionType } from "@custom-types/gatsby-plugin-transition-link"
import { TRANSITION_STATUS_ENTERING } from "@hooks/page-transition"

export const mockSocialIcons: SocialMeta = {
  email: {
    label: "Email",
    url: "email address",
    icon: "mail",
  },
  twitter: {
    label: "Twitter",
    url: "path/to/social/profile",
    icon: "twitter",
  },
  instagram: {
    label: "Instagram",
    url: "path/to/social/profile",
    icon: "instagram",
  },
  linkedin: {
    label: "Linkedin",
    url: "path/to/social/profile",
    icon: "linkedin",
  },
  dribbble: {
    label: "Dribbble",
    url: "path/to/social/profile",
    icon: "dribbble",
  },
  github: {
    label: "Github",
    url: "path/to/social/profile",
    icon: "github",
  },
}

export const mockFluidImageObject: FluidObject = {
  aspectRatio: 1.5,
  sizes: "(max-width: 1000px) 100vw, 1000px",
  src: "/static/4a7de89fcee5d7cd97c940895602fbe0/2f7e7/cover-image.jpg",
  srcSet:
    "/static/4a7de89fcee5d7cd97c940895602fbe0/4d406/cover-image.jpg 250w,↵/static/4a7de89fcee5d7cd97c940895602fbe0/32ee9/cover-image.jpg 500w,↵/static/4a7de89fcee5d7cd97c940895602fbe0/2f7e7/cover-image.jpg",
}

export const createStateMock = (animType?: TransitionType) => {
  return {
    current: {
      delay: 0,
      length: 0,
      state: {},
    },
    entry: {
      delay: 0,
      length: 0,
      state: {
        animType,
      },
    },
    exit: {
      delay: 0,
      length: 0,
      state: {
        animType,
      },
    },
    transitionStatus: TRANSITION_STATUS_ENTERING,
    mount: false,
  }
}
