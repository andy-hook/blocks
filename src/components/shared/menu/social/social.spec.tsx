import React from "react"
import { renderWithTheme } from "@test-utils"

import Social from "./social"
import { mockSocialIcons } from "@data/mocks"

describe("<Social />", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme(
      "dark",
      <Social items={mockSocialIcons} open={true} />
    )
    expect(tree).toMatchSnapshot()
  })
})
