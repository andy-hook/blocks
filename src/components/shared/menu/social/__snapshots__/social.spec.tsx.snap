// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Social /> renders correctly 1`] = `
<ul
  className="sc-htpNat gZZUDJ"
>
  <li
    className="sc-bxivhb ggjDqm"
  >
    <a
      aria-label="Email"
      className="sc-ifAKCX cxbVWF"
      href="email address"
      onClick={[Function]}
      target="_blank"
    >
      <div
        className="sc-EHOje elamHh"
      >
        <div
          className="sc-bdVaJa dqBkPn"
        >
          <svg
            className="sc-bwzfXH kdIDyY icon"
          >
            <use
              xlinkHref="/icon-sprite.svg#mail"
            />
          </svg>
        </div>
      </div>
    </a>
  </li>
  <li
    className="sc-bxivhb ggjDqm"
  >
    <a
      aria-label="Twitter"
      className="sc-ifAKCX cxbVWF"
      href="path/to/social/profile"
      onClick={[Function]}
      target="_blank"
    >
      <div
        className="sc-EHOje elamHh"
      >
        <div
          className="sc-bdVaJa dqBkPn"
        >
          <svg
            className="sc-bwzfXH kdIDyY icon"
          >
            <use
              xlinkHref="/icon-sprite.svg#twitter"
            />
          </svg>
        </div>
      </div>
    </a>
  </li>
  <li
    className="sc-bxivhb ggjDqm"
  >
    <a
      aria-label="Instagram"
      className="sc-ifAKCX cxbVWF"
      href="path/to/social/profile"
      onClick={[Function]}
      target="_blank"
    >
      <div
        className="sc-EHOje elamHh"
      >
        <div
          className="sc-bdVaJa dqBkPn"
        >
          <svg
            className="sc-bwzfXH kdIDyY icon"
          >
            <use
              xlinkHref="/icon-sprite.svg#instagram"
            />
          </svg>
        </div>
      </div>
    </a>
  </li>
  <li
    className="sc-bxivhb ggjDqm"
  >
    <a
      aria-label="Linkedin"
      className="sc-ifAKCX cxbVWF"
      href="path/to/social/profile"
      onClick={[Function]}
      target="_blank"
    >
      <div
        className="sc-EHOje elamHh"
      >
        <div
          className="sc-bdVaJa dqBkPn"
        >
          <svg
            className="sc-bwzfXH kdIDyY icon"
          >
            <use
              xlinkHref="/icon-sprite.svg#linkedin"
            />
          </svg>
        </div>
      </div>
    </a>
  </li>
  <li
    className="sc-bxivhb ggjDqm"
  >
    <a
      aria-label="Dribbble"
      className="sc-ifAKCX cxbVWF"
      href="path/to/social/profile"
      onClick={[Function]}
      target="_blank"
    >
      <div
        className="sc-EHOje elamHh"
      >
        <div
          className="sc-bdVaJa dqBkPn"
        >
          <svg
            className="sc-bwzfXH kdIDyY icon"
          >
            <use
              xlinkHref="/icon-sprite.svg#dribbble"
            />
          </svg>
        </div>
      </div>
    </a>
  </li>
  <li
    className="sc-bxivhb ggjDqm"
  >
    <a
      aria-label="Github"
      className="sc-ifAKCX cxbVWF"
      href="path/to/social/profile"
      onClick={[Function]}
      target="_blank"
    >
      <div
        className="sc-EHOje elamHh"
      >
        <div
          className="sc-bdVaJa dqBkPn"
        >
          <svg
            className="sc-bwzfXH kdIDyY icon"
          >
            <use
              xlinkHref="/icon-sprite.svg#github"
            />
          </svg>
        </div>
      </div>
    </a>
  </li>
</ul>
`;
